### PR TITLE
Update OpenAI GPT model card

### PR DESCRIPTION
**Pull request by linnettuscano**
# What does this PR do?

This PR updates the OpenAI GPT model card documentation to better reflect recent improvements and clarifications. 

## Before submitting
- [✔️ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ✔️] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [✔️] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Documentation: @stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/openai-gpt.md [hunk 1]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
 
 ## Overview
 
-OpenAI GPT model was proposed in [Improving Language Understanding by Generative Pre-Training](https://s3-us-west-2.amazonaws.com/openai-assets/research-covers/language-unsupervised/language_understanding_paper.pdf)
-by Alec Radford, Karthik Narasimhan, Tim Salimans and Ilya Sutskever. It's a causal (unidirectional) transformer
-pre-trained using language modeling on a large corpus with long range dependencies, the Toronto Book Corpus.
+The OpenAI GPT (Generative Pre-trained Transformer) model was originally proposed in the paper "Improving Language Understanding by Generative Pre-Training" by Alec Radford, Karthik Narasimhan, Tim Salimans, and Ilya Sutskever. GPT is a causal (unidirectional) transformer pre-trained on a large corpus, utilizing the Toronto Book Corpus, and fine-tuned for a variety of NLP tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:30]: The badges should go above `# OpenAI GPT` and be aligned to the right:

```md
<div style="float: right;">
   badges
</div>

Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:30]:
```suggestion
[GPT (Generative Pre-trained Transformer)](https://cdn.openai.com/research-covers/language-unsupervised/language_understanding_paper.pdf) focuses on effectively learning text representations and transferring them to tasks. This model trains the Transformer decoder to predict the next word, and then fine-tuned on labeled data.
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 2]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
...
 
-The abstract from the paper is the following:
+The core innovation of GPT lies in its ability to generate high-quality text, making it well-suited for a variety of natural language understanding tasks such as textual entailment, question answering, semantic similarity, and document classification. The model is based on a transformer architecture and trained using the language modeling objective, followed by task-specific fine-tuning.
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:32]:
```suggestion
GPT can generate high-quality text, making it well-suited for a variety of natural language understanding tasks such as textual entailment, question answering, semantic similarity, and document classification.
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 3]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
...
-labeled data for learning these specific tasks is scarce, making it challenging for discriminatively trained models to
-perform adequately. We demonstrate that large gains on these tasks can be realized by generative pretraining of a
-language model on a diverse corpus of unlabeled text, followed by discriminative fine-tuning on each specific task. In
-contrast to previous approaches, we make use of task-aware input transformations during fine-tuning to achieve
-effective transfer while requiring minimal changes to the model architecture. We demonstrate the effectiveness of our
-approach on a wide range of benchmarks for natural language understanding. Our general task-agnostic model outperforms
-discriminatively trained models that use architectures specifically crafted for each task, significantly improving upon
-the state of the art in 9 out of the 12 tasks studied.*
+In comparison to previous discriminative models, GPT achieves significant gains by utilizing a generative pretraining approach on a large, diverse corpus of unlabeled text, followed by fine-tuning for specific tasks. It achieves state-of-the-art results in a variety of benchmarks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:34]: We can remove this paragraph

File: docs/source/en/model_doc/openai-gpt.md [hunk 4]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
...
 
-[Write With Transformer](https://transformer.huggingface.co/doc/gpt) is a webapp created and hosted by Hugging Face
-showcasing the generative capabilities of several models. GPT is one of them.
+Key Improvements and Features:
+Scalability: The model can scale efficiently with larger training datasets, improving performance on a wide range of NLP tasks.
+
+Generative Pre-training: GPT leverages unsupervised pretraining on large text corpora, making it task-agnostic and highly adaptable.
+
+Fine-Tuning Flexibility: After pretraining, GPT can be fine-tuned for specific tasks with minimal changes to the model architecture.
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:41]:
```suggestion
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 5]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
...
+
+
+Official checkpoints can be found on the [Hugging Face Model Hub](https://huggingface.co/models).
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:44]:
```suggestion
You can find all the original GPT checkpoints under the [OpenAI community](https://huggingface.co/openai-community/openai-gpt) organization.

> [!TIP]
> Click on the GPT models in the right sidebar for more examples of how to apply GPT to different language tasks.

The example below demonstrates how to generate text with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 6]
```diff
@@ -27,76 +27,262 @@ rendered properly in your Markdown viewer.
...
 
-This model was contributed by [thomwolf](https://huggingface.co/thomwolf). The original code can be found [here](https://github.com/openai/finetune-transformer-lm).
 
 ## Usage tips
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:47]: We can remove everything else after this as its out-of-scope and a bit overwhelming. Take a look at the [Llama](https://huggingface.co/docs/transformers/main/en/model_doc/llama) doc for how this section should be formatted.

It should basically look like the example below. Include code snippets for using GPT with `Pipeline`, `AutoModel` and from the command line. Then add a `## Notes` section after it.

```md
<hfoptions id="usage">
<hfoption id="Pipeline">
...
...
...
</hfoption>
</hfoptions>

## Notes
```

**Review by stevhliu — COMMENTED** [review]
Great start! However, I think its a bit too much info so we should take the extra stuff out and only add the really important stuff, which is the code examples demonstrating the `Pipeline`, `AutoModel` and `transformers-cli`.

File: docs/source/en/model_doc/openai-gpt.md (hunk 1 above)
Comment (linnettuscano) [reply docs/source/en/model_doc/openai-gpt.md]: done

File: docs/source/en/model_doc/openai-gpt.md (hunk 3 above)
Comment (linnettuscano) [reply docs/source/en/model_doc/openai-gpt.md]: done

File: docs/source/en/model_doc/openai-gpt.md (hunk 6 above)
Comment (linnettuscano) [reply docs/source/en/model_doc/openai-gpt.md:40]: done

**Comment by linnettuscano** [comment]
@stevhliu i've made all the changes as you suggested. please check and let me know if i should modify anything else.

File: docs/source/en/model_doc/openai-gpt.md [hunk 7]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
+</hfoption> <hfoption id="AutoModel">
 
-## Resources
+You can also load the model using `AutoModelForCausalLM` :
+```python
+from transformers import AutoModelForCausalLM, AutoTokenizer
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with OpenAI GPT. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+tokenizer = AutoTokenizer.from_pretrained("openai-gpt")
+model = AutoModelForCausalLM.from_pretrained("openai-gpt")
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:62]:
```suggestion
model = AutoModelForCausalLM.from_pretrained("openai-community/openai-gpt", torch_dtype=torch.float16)
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 8]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
 
-<PipelineTag pipeline="text-classification"/>
+inputs = tokenizer("The future of AI is", return_tensors="pt")
+outputs = model.generate(**inputs, max_length=50)
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+```
 
-- A blog post on [outperforming OpenAI GPT-3 with SetFit for text-classification](https://www.philschmid.de/getting-started-setfit).
-- See also: [Text classification task guide](../tasks/sequence_classification)
+<hfoption id="transformers-cli">
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:69]:
```suggestion
</hfoption>
<hfoption id="transformers-cli">
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 9]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
+</hfoptions>
 
-<PipelineTag pipeline="token-classification"/>
+## Notes
 
-- A course material on [Byte-Pair Encoding tokenization](https://huggingface.co/course/en/chapter6/5).
+- This model was trained using a causal language modeling (CLM) objective.
+- It uses byte-level BPE tokenization.
+- Right-side padding is recommended due to absolute position embeddings.
+- Best suited for text generation and zero-shot transfer learning.
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:83]:
```suggestion
- Inputs should be padded on the right because GPT uses absolute position embeddings.
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 10]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
 
-If you want to reproduce the original tokenization process of the *OpenAI GPT* paper, you will need to install `ftfy`
-and `SpaCy`:
+```python
+from transformers import pipeline
 
-```bash
-pip install spacy ftfy==4.4.3
-python -m spacy download en
+generator = pipeline("text-generation", model="openai-gpt")
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:50]:
```suggestion
generator = pipeline(task="text-generation", model="openai-community/gpt", torch_dtype=torch.float16, device=0)
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 11]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-Note:
+You can use this model with the `pipeline` API :
 
-If you want to reproduce the original tokenization process of the *OpenAI GPT* paper, you will need to install `ftfy`
-and `SpaCy`:
+```python
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:48]:
```suggestion
import torch
from transformers import pipeline
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 12]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
-using BERT's `BasicTokenizer` followed by Byte-Pair Encoding (which should be fine for most usage, don't worry).
+</hfoption> <hfoption id="AutoModel">
 
-## Resources
+You can also load the model using `AutoModelForCausalLM` :
+```python
+from transformers import AutoModelForCausalLM, AutoTokenizer
 
-A list of official Hugging Face and community (indicated by 🌎) resources to help you get started with OpenAI GPT. If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+tokenizer = AutoTokenizer.from_pretrained("openai-gpt")
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:61]:
```suggestion
tokenizer = AutoTokenizer.from_pretrained("openai-community/gpt")
```

File: docs/source/en/model_doc/openai-gpt.md [hunk 13]
```diff
@@ -14,154 +14,122 @@ rendered properly in your Markdown viewer.
...
+print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+```
 
-- A blog post on [outperforming OpenAI GPT-3 with SetFit for text-classification](https://www.philschmid.de/getting-started-setfit).
-- See also: [Text classification task guide](../tasks/sequence_classification)
+<hfoption id="transformers-cli">
 
-<PipelineTag pipeline="text-generation"/>
+```bash
+echo -e "The future of AI is" | transformers-cli run --task text-generation --model openai-gpt --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/openai-gpt.md:72]:
```suggestion
echo -e "The future of AI is" | transformers-cli run --task text-generation --model openai-community/openai-gpt --device 0
```

**Comment by linnettuscano** [comment]
Hi @stevhliu , I've made the suggested changes and also added the missing badges. Let me know if there's anything else I should update. Thanks!

**Review by stevhliu — APPROVED** [review]
Nice job, thanks again! 🎉
